### ISSUE TITLE
Remove support for 1.8

### ISSUE THREAD

**Issue by xaviershay** [rspec/rspec-core#2124]
- Search the code base for any switches or conditionals supporting 1.8 and remove them.
- Remove from travis.

**Comment by xaviershay** [comment]
Should we remove 1.9 support as well? By the time this gets out it will have been EOLed for a year: https://www.ruby-lang.org/en/news/2015/02/23/support-for-ruby-1-9-3-has-ended/

I vote yes.

**Comment by JonRowe** [comment]
I vote no, two version jumps would be a bad idea in my mind, especially as 1.9 is still (I believe) the vendored ruby for a LTS version of ubuntu..

**Comment by samphippen** [comment]
If we were to do this right now I'd be :-1: on the removing of 1.9. We're not shipping 4.0 yet, and so I don't think it's meaningful to talk about what versions it will support. We should consult people on 1.9 as and when we're a few months out from shipping.

**Comment by e2** [comment]
TL;DR - 1.8 and 1.9 don't compare. Dropping 1.8 is a major risk/pain. Dropping 1.9, 2.0 and even 2.1 is pretty much just a matter of how convenient it is for users to install a newer Ruby. And it's recommended too, anyway. RSpec is also a development tool, not a core server library - so it makes sense to have it running on the latest usable Ruby possible.

Just my 2 cents on this:

The shift from Ruby 1.8 to 1.9 is enormous. Meanwhile, going from 1.9 to even 2.3.1 is usually painless. So there's nothing indicative in the version number.

That's why I'm guessing that even today more people are going to be stuck with 1.8 than on 1.9 (because it's so much easier to upgrade, it's almost a no-brainer).

So dropping 1.8 is a big thing, though long overdue. The Ruby community doesn't deserve to keep supporting it as baggage, so it's a huge relief to drop it.

Meanwhile, 1.9 is basically 2.x with a misleading name. Dropping it would be as trivial as dropping 2.0.0 on a project that supports 2.1.

As of today, Ruby >= 2.2.5 is officially recommended, so even dropping 2.1 would just be a matter of convenience and nothing else much.

As for LTS support and related, I remember a post about Debian Jessie and backporting Ruby 2.2 to it.

The issue there is that there's too much recompiling and testing just for a "backport". The updates and whole operation would be too massive to qualify as a "backport".

The solution there is of course to build/download your own Ruby version and install gems so they build with the ABI for whatever Ruby you want.

With that said, RSpec is a special case, because it's the primary tool needed for upgrading. If you have an old Ruby 1.8 project, using the newest RSpec and transpec would definitely help get the specs updated sooner.

So RSpec is actually a tool to help drop support for older Rubies ;) 

But, as a development tool, it's usually no big deal (in business sense) to stop running RSpec on an old Ruby to test the code of an old codebase. You can upgrade Ruby during development and upgrading the codebase, and then downgrade again to 1.9 by locking to an earlier RSpec.

RSpec 2.99 had awesome work put into it to make upgrading painless - and it works the other way too (downgrading).

So practically, dropping Ruby 1.9 would be no big deal. It's not like a 1.9 codebase needs to be tested by a test framework compatible with Ruby 1.9. Meanwhile, life is too precious to realize you've spent the last 3 hours debugging a Ruby 1.9 bug that will never be fixed.

I'd consider dropping support for everything up to Ruby 2.1 in 4.x. (The mistake they made with Debian Jessie is that they didn't switch to Ruby 2.2 before releasing, although they could have).

**Comment by xaviershay** [comment]
Not going to keep this open. There's no ongoing pain here, if it begins causing some we can talk about it again.
